Jenkinsfile.cloud

```diff
@@ -55,20 +81,32 @@ node(env.NODE) {
         """
     }
 
-    def commit, dirpath, qcow
+    def commit, version, dirpath, qcow, vmdk
     stage("Finalizing") {
         def image = sh(returnStdout: true, script: "ls /var/lib/imagefactory/storage/*.body").trim()
         // just introspect after the fact to avoid race conditions
         commit = sh(returnStdout: true,
                     script: """LIBGUESTFS_BACKEND=direct virt-cat -a ${image} -m /dev/coreos/root:/ \
                                     /ostree/repo/refs/remotes/rhcos/${ref}""").trim()
-        currentBuild.description = 'commit ' + commit
+
+        // do this again *after* running imgfac to close race
+        // fetch parent too for the later db diff
+        sh "ostree --repo=repo pull --depth=1 --commit-metadata-only tmp ${commit}"
+        version = sh(returnStdout: true,
+                     script: "ostree show --repo=repo --print-metadata-key=version ${commit}").trim()
+        assert (version.startsWith("'") && version.endsWith("'"))
+        version = version[1..-2] // trim single quotes
+
+        currentBuild.description = "${version} (${commit})"
 
         def dirname = (new Date()).format("YYYY-MM-dd-HH-mm-ss-") + commit
```

Comment:
One tricky bit here is that I left the dirname alone instead of using the version string as before. The reason is that if we force-nocache to rebuild the same version, I still wanted it to be output in a new directory instead of overwriting.